MusiciansAPP/MusiciansAPP.BL/ArtistsService/Logic/TrackDataService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using AutoMapper;
+using MusiciansAPP.BL.ArtistsService.BLModels;
+using MusiciansAPP.BL.ArtistsService.Interfaces;
+using MusiciansAPP.DAL.DBDataProvider.Interfaces;
+using MusiciansAPP.Domain;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace MusiciansAPP.BL.ArtistsService.Logic;
+
+public class TrackDataService : ITrackDataService
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IMapper _mapper;
+
+    public TrackDataService(IUnitOfWork unitOfWork, IMapper mapper)
+    {
+        _unitOfWork = unitOfWork;
+        _mapper = mapper;
+    }
+
+    public async Task SaveTopTracksAsync(ArtistTracksBL tracks)
```

Comment:
The logic of this method (and all other similar methods) is just to work with DB.
We don't need to use TrackDataService. We can move these pieces of logic directly to UnitOfWork and call it instead of TrackDataService.
UnitOfWork is a pattern that was designed to encapsulate DB logic and call different specific repositories.